extensions/panache/panache-common/deployment/src/main/java/io/quarkus/panache/common/deployment/PanacheRepositoryEnhancer.java

```diff
@@ -101,26 +117,102 @@ private String recursivelyFindEntityTypeFromClass(DotName clazz, DotName reposit
-                if (!JandexUtil.containsMethod(daoClassInfo, method)) {
+                String descriptor = JandexUtil.getDescriptor(method, name -> typeArguments.get(name));
+                if (!userMethods.contains(method.name() + "/" + descriptor)) {
                     AnnotationInstance bridge = method.annotation(JandexUtil.DOTNAME_GENERATE_BRIDGE);
                     if (bridge != null) {
-                        generateMethod(method, bridge.value("targetReturnTypeErased"));
+                        generateModelBridge(method, bridge.value("targetReturnTypeErased"));
+                        if (needsJvmBridge(method)) {
+                            generateJvmBridge(method);
+                        }
                     }
                 }
             }
             super.visitEnd();
         }
 
-        private void generateMethod(MethodInfo method, AnnotationValue targetReturnTypeErased) {
-            String descriptor = JandexUtil.getDescriptor(method, name -> name.equals("Entity") ? entitySignature : null);
-            String signature = JandexUtil.getSignature(method, name -> name.equals("Entity") ? entitySignature : null);
+        private boolean needsJvmBridge(MethodInfo method) {
+            if (needsJvmBridge(method.returnType()))
+                return true;
+            for (org.jboss.jandex.Type paramType : method.parameters()) {
+                if (needsJvmBridge(paramType))
+                    return true;
+            }
+            return false;
+        }
+
+        private boolean needsJvmBridge(org.jboss.jandex.Type type) {
+            if (type.kind() == Kind.TYPE_VARIABLE) {
+                String typeParamName = type.asTypeVariable().identifier();
+                return typeArguments.containsKey(typeParamName);
+            }
+            return false;
+        }
+
+        private void generateJvmBridge(MethodInfo method) {
+            // get a bounds-erased descriptor
+            String descriptor = JandexUtil.getDescriptor(method, name -> null);
+            // make sure we need a bridge
+            if (!userMethods.contains(method.name() + "/" + descriptor)) {
+                MethodVisitor mv = super.visitMethod(Opcodes.ACC_PUBLIC | Opcodes.ACC_SYNTHETIC | Opcodes.ACC_BRIDGE,
+                        method.name(),
+                        descriptor,
+                        null,
+                        null);
+                List<org.jboss.jandex.Type> parameters = method.parameters();
+                for (int i = 0; i < parameters.size(); i++) {
+                    mv.visitParameter(method.parameterName(i), 0 /* modifiers */);
+                }
+                mv.visitCode();
+                // this
+                mv.visitIntInsn(Opcodes.ALOAD, 0);
+                // each param
+                for (int i = 0; i < parameters.size(); i++) {
+                    org.jboss.jandex.Type paramType = parameters.get(i);
+                    mv.visitIntInsn(Opcodes.ALOAD, i + 1);
```

Comment:
I don't really understand the context here, but are you sure there will never be primitive params?

Reply:
ATM there are none. I've added a check if it ever happens.